### PR TITLE
Unit test for students group service

### PR DESCRIPTION
**Pull request by BohdanBBG**
Added unit-tests for StudentGroupService methods:
  - CreateStudentGroupAsync()
  - UpdateStudentGroupAsync()

### CODE REVIEW

File: CharlieBackend.Api/Controllers/StudentsController.cs [hunk 1]
```diff
@@ -100,6 +100,22 @@ public async Task<ActionResult<IList<StudentDto>>> GetAllActiveStudents()
             return Ok(studentsModels);
         }
 
+        /// <summary>
```
Comment (SerjKocherga) [inline CharlieBackend.Api/Controllers/StudentsController.cs:103]: Why these changes are here?
I thought this branch is for unit-tests task, not for documentation and below changes in services and repository.
Please, separate it.

**Review by SerjKocherga — CHANGES_REQUESTED** [review]

File: CharlieBackend.Api/Controllers/StudentsController.cs (hunk 1 above)
Comment (SerjKocherga) [reply CharlieBackend.Api/Controllers/StudentsController.cs:103]: Also, please, add description for your PR.

Comment (BohdanBBG) [reply CharlieBackend.Api/Controllers/StudentsController.cs:103]: Some changes are taken from another PR. I merged this branch from my earlier PR. This PR is about the last 3 commits

**Review by SerjKocherga — APPROVED** [review]

File: CharlieBackend.Api.UnitTes/StudentGroupServiceTest.cs [hunk 2]
```diff
@@ -0,0 +1,241 @@
...
+
+            studentGroupRepositoryMock.Setup(x => x.IsGroupNameExistAsync(newStudentGroup.Name))
+                    .ReturnsAsync(false);
+
+            studentGroupRepositoryMock.Setup(x => x.IsGroupNameExistAsync(existingStudentGroup.Name))
+                    .ReturnsAsync(true);
+
+            _unitOfWorkMock.Setup(x => x.StudentGroupRepository).Returns(studentGroupRepositoryMock.Object);
+            _unitOfWorkMock.Setup(x => x.StudentRepository).Returns(studentsRepositoryMock.Object);
+            _unitOfWorkMock.Setup(x => x.MentorRepository).Returns(mentorsRepositoryMock.Object);
```
Comment (x1mx1x) [inline CharlieBackend.Api.UnitTes/StudentGroupServiceTest.cs:104]: you can divide it into several lines

File: CharlieBackend.Api.UnitTes/StudentGroupServiceTest.cs [hunk 3]
```diff
@@ -0,0 +1,241 @@
...
+            studentGroupRepositoryMock.Setup(x => x.UpdateManyToMany(It.IsAny<IList<StudentOfStudentGroup>>(),
+                                                                     It.IsAny<IList<StudentOfStudentGroup>>()));
+
+            mentorsRepositoryMock.Setup(x => x.UpdateMentorGroups(It.IsAny<IList<MentorOfStudentGroup>>(),
+                                                                  It.IsAny<IList<MentorOfStudentGroup>>()));
+
+            _unitOfWorkMock.Setup(x => x.StudentGroupRepository).Returns(studentGroupRepositoryMock.Object);
+            _unitOfWorkMock.Setup(x => x.StudentRepository).Returns(studentsRepositoryMock.Object);
+            _unitOfWorkMock.Setup(x => x.MentorRepository).Returns(mentorsRepositoryMock.Object);
+            _unitOfWorkMock.Setup(x => x.CourseRepository).Returns(courseRepositoryMock.Object);
```
Comment (x1mx1x) [inline CharlieBackend.Api.UnitTes/StudentGroupServiceTest.cs:207]: too much symbols in the lines

File: CharlieBackend.Api.UnitTes/StudentGroupServiceTest.cs [hunk 4]
```diff
@@ -0,0 +1,241 @@
...
+                _mapper,
+                _loggerMock.Object
+                );
+
+            //Act
+
+            var successResult = await studentGroupService.UpdateStudentGroupAsync(existingStudentGroup.Id, updateStudentGroupDto);
+            var groupNameExistResult = await studentGroupService.UpdateStudentGroupAsync(existingStudentGroup.Id, existingNameStudentGroupDto);
+            var groupNotExistResult = await studentGroupService.UpdateStudentGroupAsync(notExistingGroupId, updateStudentGroupDto);
+            var nullGroupResult = await studentGroupService.UpdateStudentGroupAsync(existingStudentGroup.Id, null);
```
Comment (x1mx1x) [inline CharlieBackend.Api.UnitTes/StudentGroupServiceTest.cs:220]: too much symbols in the lines

**Review by x1mx1x — APPROVED** [review]

**Review by SnipeFrost — APPROVED** [review]

File: CharlieBackend.Api.UnitTes/StudentGroupServiceTest.cs (hunk 2 above)
Comment (BohdanBBG) [reply CharlieBackend.Api.UnitTes/StudentGroupServiceTest.cs:104]: It is divided into same logical block, spaces can harm the logical structure of the code

File: CharlieBackend.Api.UnitTes/StudentGroupServiceTest.cs (hunk 3 above)
Comment (BohdanBBG) [reply CharlieBackend.Api.UnitTes/StudentGroupServiceTest.cs:207]: It is divided into same logical block, spaces can harm the logical structure of the code

File: CharlieBackend.Api.UnitTes/StudentGroupServiceTest.cs (hunk 4 above)
Comment (BohdanBBG) [reply CharlieBackend.Api.UnitTes/StudentGroupServiceTest.cs:220]: It is divided into same logical block, spaces can harm the logical structure of the code

**Review by Vlad1023 — APPROVED** [review]

**Review by sober607 — APPROVED** [review]
